pkg/crm/crm_appinst.go

```diff
@@ -68,6 +69,11 @@ func (s *CRMData) appInstChanged(ctx context.Context, old *edgeproto.AppInst, ne
 				new.Fields = edgeproto.AppInstAllFields
 			} else {
 				new.Fields = old.GetDiffFields(new).Fields()
+				oldDNS, ok := new.Annotations[cloudcommon.AnnotationPreviousDNSName]
+				if ok && oldDNS == old.Uri {
+					_ = s.AppInstDNSChanged(ctx, s.cloudletKey, old, new, responseSender)
```

Comment:
This still depends on old so this logic won't translate to the CCRM, I think it would be more straightforward to just check
```
if old.Uri != new.Uri {
```
In the CCRM, we don't have to worry about multiple updates via notify that could erroneously trigger another DNS update. In the CCRM if we get a call to Update the DNS, it's because the Controller decided it needed to be done. We just need the old URI to be able to delete the old DNS entry, not to determine whether or not to perform the action.

Reply:
Yeah, I didn't make this change yet - I pushed some intermediate changeset, which didn't address your comments about moving the logic into controller-data.go yet. Will fix.

Reply:
Updated the code to handle the transitions in the `controller-data.go` instead so CCRM can use the same logic.